src/fs.rs

```diff
@@ -561,6 +561,20 @@ impl Directory {
     }
 }
 
+// show file url as relative to static path
+macro_rules! encode_file_url {
```

Comment:
Why macros is necessary?

Reply:
It is a macro sanitize, so as not to repeat myself. It is necessary to limit the lifetime of the strings within the scope of the `if`. It has been embedded in the commit because it is in the same logic.

Reply:
Couldn't it be done as inline function?

Reply:
With this implementation I still have problems in the life time of the borrowed values
```
// show file url as relative to static path
#[inline]
fn encode_file_url<'a>(path: Cow<'a, str>) -> PercentEncode<'a, DEFAULT_ENCODE_SET> {
    utf8_percent_encode(&path, DEFAULT_ENCODE_SET)
}

// " -- &quot;  & -- &amp;  ' -- &#x27;  < -- &lt;  > -- &gt;  / -- &#x2f;
#[inline]
fn encode_file_name<'a>(name: Cow<'a, str>) -> Escaped<'a> {
    escape_html_entity(&name)
}
```

Reply:
Does it complain because of of intermediate function there?

Reply:
Exactly, in the borrow path and borrow name with the typical error:
```
escape_html_entity(&name)                                                     
                   ^^^^ borrowed value does not live long enough  
```

Reply:
Oh, ok, let's leave it as it is then

Reply:
Ok thanks. My first commit to actix-web, it will not be the last one :smiley: